src/ip/tools.cc

```diff
@@ -55,8 +56,21 @@ Ip::ProbeTransport()
     debugs(3, 2, "Missing RFC 3493 compliance - attempting split IPv4 and IPv6 stacks ...");
     EnableIpv6 |= IPV6_SPECIAL_SPLITSTACK;
 #endif
-    // TODO: attempt to use the socket to connect somewhere ?
-    //  needs to be safe to contact and with guaranteed working IPv6 at the other end.
+
+    // Test for IPv6 loopback/localhost address binding
+    Ip::Address ip;
+    ip.setLocalhost();
+    if (ip.isIPv6()) { // false on --disable-ipv6 builds
+        struct sockaddr_in6 sin;
+        ip.getSockAddr(sin);
+        if (bind(s, reinterpret_cast<struct sockaddr *>(&sin), sizeof(sin)) != 0) {
+            debugs(3, DBG_CRITICAL, "WARNING: BCP 177 violation. Detected non-functional IPv6 loopback.");
```

Comment:
Since "non-functional IPv6 loopback" does not necessarily break anything, perhaps this should be `DBG_IMPORTANT`? I am just asking and not even suggesting a change here. Your call.

Reply:
I picked critical because this system is claiming IPv6 to be enabled (socket succeeded) but lacking any NIC registered as loopback. There are quite a number of connections which use loopback/localhost for security purposes. The IPv4 localhost is less secure than IPv6 loopback in VM and container environments. As shown in the bug report though it might be intentional so the message is a warning and reversion to IPv4 anyway.

Reply:
FWIW, I do not see the connection between some systems using loopback for security purposes and Squid claiming that not using loopback is a critical problem (presumably for Squid). Clearly, Squid could check lots of BCPs, but (thankfully) does not, because Squid is a proxy, not a BCP compliance validator.

Said that, I am resolving this minor comment. It is your call.